docs/docs/migration/migration-from-v6-to-v7.md

```diff
@@ -165,9 +175,357 @@ Make sure to use a meaningful strategy name as this name can be used to identify
+
+Remove the following env vars from the API
+
+```diff
+//.env
+
+- DAM_CDN_ENABLED=
+- DAM_CDN_DOMAIN=
+- DAM_CDN_ORIGIN_HEADER=
+- DAM_DISABLE_CDN_ORIGIN_HEADER_CHECK=false
+```
+
+If you want to enable the origin check:
+
+1. Set the following env vars for the API
+
+    ```diff
+    // .env
+
+    + CDN_ORIGIN_CHECK_SECRET="Use value from DAM_CDN_ORIGIN_HEADER to avoid downtime"
+    ```
+
+    _environment-variables.ts_
+
+    ```diff
+    // environment-variables.ts
+
+    + @IsString()
+    + @ValidateIf(() => process.env.NODE_ENV === "production")
+    + CDN_ORIGIN_CHECK_SECRET: string;
+    ```
+
+    ```
+    // config.ts
+
+    + cdn: {
+    +  originCheckSecret: envVars.CDN_ORIGIN_CHECK_SECRET,
+    + },
+    ```
+
+2. Add CdnGuard
+
+    ```diff
+    // main.ts
+
+    + // if CDN is enabled, make sure all traffic is either coming from the CDN or internal sources
+    + if (config.cdn.originCheckSecret) {
+    +   app.useGlobalGuards(new CdnGuard({ headerName: "x-cdn-origin-check", headerValue: config.cdn.originCheckSecret }));
+    + }
+    ```
+
+3. Adjust `site/server.js`
+
+```diff
+// site/server.js
+
+- const cdnEnabled = process.env.CDN_ENABLED === "true";
+- const disableCdnOriginHeaderCheck = process.env.DISABLE_CDN_ORIGIN_HEADER_CHECK === "true";
+- const cdnOriginHeader = process.env.CDN_ORIGIN_HEADER;
++ const cdnOriginCheckSecret = process.env.CDN_ORIGIN_CHECK_SECRET;
+
+// ...
+
+- if (cdnEnabled && !disableCdnOriginHeaderCheck) {
+-    const incomingCdnOriginHeader = req.headers["x-cdn-origin-check"];
+-    if (cdnOriginHeader !== incomingCdnOriginHeader) {
++ if (cdnOriginCheckSecret) {
++    if (req.headers["x-cdn-origin-check"] !== cdnOriginCheckSecret) {
+```
+
+4. DNS changes might be required. `api.example.com` should point to CDN, CDN should point to internal API domain
+
+### Remove `language` field from `User`
+
+```diff
+// static-users.ts
+
+export const staticUsers = {
+    admin: {
+        id: "3b09cc12-c7e6-4d16-b858-40a822f2c548",
+        name: "Admin",
+        email: "admin@customer.com",
+-       language: "en",
+    },
+    // ...
+} satisfies Record<string, User>;
+```
+
+### Remove `@PublicApi()` and rename `@DisableGlobalGuard()`
+
+Replace all usages of `@PublicApi()` and `@DisableGlobalGuard()` with `@DisableCometGuards()`.
+Use this occasion to check if all operations decorated with this decorator **should actually be public and don't return any confidential data**.
+
+### Remove unnecessary dependencies
+
+Remove following dependencies **if you don't use them in your project**:
+
+-   `@aws-sdk/client-s3`
+-   `@azure/storage-blob`
+-   `pg-error-constants`
+
+### API Generator: Remove support for `visible` boolean, use `status` enum instead
+
+Replace the `visible` boolean field with a `status` enum field.
+Recommended enum values are (depending on the use case):
+
+-   Published/Unpublished
+-   Visible/Invisible
+-   Active/Deleted
+-   Active/Archived
```

Comment:
Also Recommended:
- Published/Unpublished/Archived
- Published/Unpublished/Deleted

and I think we don't want to use visible/invisible, that's the same as Published/Unpublished

Reply:
https://github.com/vivid-planet/comet/pull/2117/commits/896b760a0c663f77c1340c5dbdbd70f782dcc176